src/types/ApplicationForm.ts

```diff
@@ -13,6 +14,7 @@ export interface ApplicationForm {
   readonly id: number;
   opportunityId: number;
   version: number;
+  externalId: string | null;
```

Comment:
The commit message says 

> Supplying the field is mandatory (for consistency) but the value may be `null`.

Can you please expand on that a bit? What is this consistent with, and why is that consistency desirable? `fields`, below, is both optional and nullable.

Reply:
Will do in the comments in the code.

Reply:
That will be good, but I did mean it as a genuine question, not only - or even primarily - as a nudge!

Reply:
1. The `application_forms` table has the `external_id` column (unlike `fields` which is another relation via join).
2. We don't want many types for `ApplicationForm` if we can avoid it.
3. I am not certain that the `externalId` will always be available.
4. The `externalId` is for convenience more than a requirement for PDC to work.

Then again, we do not always have the original order, but we mandate the original order. I am now leaning toward requiring the value even if we make one up on the spot. Having too many optional fields can be a pain.